### ISSUE TITLE
hi, if I want to use batch norm instead of instance norm, how can I implement the once-for-all part?

### ISSUE THREAD

**Issue by leonardodora** [mit-han-lab/gan-compression#65]
could you please give an example?

**Comment by lmxyy** [comment]
I think you could refer to our implemented [SuperSynchronizedBatchNorm2d module](https://github.com/mit-han-lab/gan-compression/blob/df4b89b618ea8e05f349753aefd4aa85d569c08b/models/modules/super_modules.py#L88). Remember to calibrate the batch normalization before the search.

**Comment by leonardodora** [comment]
> I think you could refer to our implemented [SuperSynchronizedBatchNorm2d module](https://github.com/mit-han-lab/gan-compression/blob/df4b89b618ea8e05f349753aefd4aa85d569c08b/models/modules/super_modules.py#L88). Remember to calibrate the batch normalization before the search.

thanks for your reply. here are two more things i want to ask.
1. what is the purpose of calibrate?
2.  SuperSynchronizedBatchNorm2d is also work for single gpu?

**Comment by lmxyy** [comment]
Different sub-networks share the same BatchNorm layers, but actually, their `running_mean` and `running_var` are different. So you need to calibrate the BatchNorm to get the correctrunning_mean` and `running_var`. The SuperSynchronizedBatchNorm2d module also works for a single GPU. In the single GPU case, it behaves exactly the same as the BatchNorm.

**Comment by leonardodora** [comment]
> Different sub-networks share the same BatchNorm layers, but actually, their `running_mean` and `running_var` are different. So you need to calibrate the BatchNorm to get the correctrunning_mean`and`running_var`. The SuperSynchronizedBatchNorm2d module also works for a single GPU. In the single GPU case, it behaves exactly the same as the BatchNorm.

thanks for your reply, i will try
